BrightId flow (UI, not functionality)

ryancreatescopy opened the pull request:
**NOTE**: for local dev, configure `VUE_APP_USER_REGISTRY_TYPE=brightid`

- [x] @samajammin to get to a state where we can merge this PR w/ UI implemented
- [ ] @samajammin create issues for remaining TODOs as BrightID epic

https://www.notion.so/efdn/Set-up-BrightID-flow-UI-and-routing-50-af0b4087079844719214f05e119a6ef9
- [x] Create routes for registering with BrightId
- [ ] Check if user registry is full before allowing user to go through verification - close out if full (just like /join flow)
- [ ] Check if contribution phase is over before allowing user to go through verification - close out if past (just like /join flow)
- [ ] Share flow with BrightID team for feedback

**Status:** The flow UI is pretty much done, just needs review. 

**Todos, but maybe for other PRs:**

Plugging in actual logic to components
- [ ] add logic to check status and ensure verification component reflects status
  - [ ] Confirm we can fetch if user registry is full / filling up - may need to add view function to registry contract
- [ ] logic to move users through the process (+ display toast messages) if user has already completed certain steps by virtue of using clr or gitcoin before: (see different states here: https://www.figma.com/file/gnA82CTLGjk0TfFHMvgyrK/?node-id=595%3A6310)
- [ ] ensure "pending" loader on verification step displays as pending while user is unverified but has started the process (e.g. looked at the screen)  - see Figma screen: https://www.figma.com/file/gnA82CTLGjk0TfFHMvgyrK/CLR?node-id=1652%3A5810
- [ ] Check for any other hardcoded logic :)

Update content
- [ ] update transaction widget with associated success/error states (see #60)
- [ ] add info about ways to verify (atm it's just a shell, it needs updating to include links to BrightID zoom party schedule, our event, etc.) on /setup/get-verified/verification
- [ ] add external links for google meet / learn to /setup landing page
- [ ] ways to verify accordion needs fixing (currently all accordions open/close at same time) - create an accordion component for this

Commit:
Sets up flow: landing -> get verified

Commit:
set up skeleton for sponsorship transaction UI

Commit:
added connect step

Commit:
some fallback if user gets to connect step without connecting

Commit:
moves brightID interface to right file

Commit:
added success state for connected step

ryancreatescopy commented:
T 👏 S 👏  L 👏 I 👏 N 👏 T 👏 I 👏 N👏 G

samajammin committed:
Fix console errors

Commit:
added verification conditional content

Commit:
added content to register and added success page

Commit:
adding todo: confetti

Commit:
added accordions and updated copy on setup page

Commit:
fixed mobile view for connect phase

Commit:
display eth balance for transactions only involving eth

Commit:
separated ButtonRow logic between Join + Setup

Think this might cause a bug: the join page doesn't seem to load straight away...

samajammin commented on vue-app/src/router/index.ts:
```diff
@@ -66,6 +69,21 @@ const routes = [
+    component: SetupLanding,
+  },
+  {
+    path: '/setup/get-verified/:step',
```
@ryancreatescopy curious on your reasoning for /get-verified/ subroute - is the thinking there will be other /setup/ routes, e.g. for l2?

ryancreatescopy commented:
Yep, exactly that.

Sent from my iPhone

> On 11 May 2021, at 20:08, Sam Richards ***@***.***> wrote:
> 
> ﻿
> @samajammin commented on this pull request.
> 
> In vue-app/src/router/index.ts:
> 
> > @@ -66,6 +69,21 @@ const routes = [
>      name: 'join',
>      component: JoinLanding,
>    },
> +  {
> +    path: '/setup',
> +    name: 'setup',
> +    component: SetupLanding,
> +  },
> +  {
> +    path: '/setup/get-verified/:step',
> @ryancreatescopy curious on your reasoning for /get-verified/ subroute - is the thinking there will be other /setup/ routes, e.g. for l2?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

Commit:
added link to sybil resistance helper article

ryancreatescopy commented:
re: previous commit. More info: https://www.notion.so/efdn/CLR-help-articles-58e3630d9fa045e39f74cadb51e74009

wackerow committed:
ts linting

wackerow committed:
adds missing getter to cart

wackerow committed:
getter name typo correction

wackerow committed:
SetupLanding layout tweaks

adds `connectOnly` boolean prop to `WalletWidget` to utilize this component for just the "Connect" button features in places other than the `NavBar`

wackerow committed:
linting

wackerow committed:
Update ProofOfIndividuality.vue

adds bright-id icon to mobile page
replaced furthestStep logic with checks for each step

wackerow committed:
ProgressBar now dynamic

Will fill width with appropriate number of step pills, depending on `totalSteps` prop

wackerow committed:
Merge branch 'develop' into bright-id-flow

wackerow committed:
use `FormProgressWidget` instead of `ButtonRow`

wackerow committed:
check for full registry

wackerow committed:
Update App.vue

renamed variable to excludedRoutes when appropriate

wackerow committed:
isVerified getter

v-else tidy

ryancreatescopy commented:
Updated description with status/progress

wackerow committed:
logic adjustment

Displays warning if round is over or full, but still allows user to progress through BrightID verification from this page

samajammin committed:
Minor cleanup

samajammin committed:
Merge branch 'develop' into bright-id-flow

samajammin committed:
Simplify BrightID routes to /verify

samajammin committed:
Fix blocking errors

samajammin committed:
Fix build

samajammin reviewed (approved):
Merging without addressing TODOs in PR description - will create separate issue for those.
